docs/spec_draft.md

```diff
@@ -705,6 +706,57 @@ lexicographic comparison on the (real, imaginary) pairs.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.multiply
+
+### Semantics
+
+Performs element-wise product of two tensors `lhs` and `rhs` and produces a
+`result` tensor. For integer element types, if the element-wise product has an
+unsigned/signed overflow/underflow, the result is implementation-defined and one
+of the following:
+
+  * mathematical result modulo $2^n$, where n is the bit width of the result,
+  for unsigned overflow/underflow. For signed integer overflow/underflow, wraps
+  the result around the representable range $[-2^{n-1},\ \ 2^{n-1} - 1]$.
+  * saturation to $2^{n-1} - 1$ (or $-2^{n-1}$) for signed overflow (or signed
+  underflow) and saturation to $2^n - 1$ (or $0$) for unsigned overflow (or
+  unsigned underflow).
+
+For floating-point element types, it implements the `multiplication` operation from
+the IEEE-754 specification.
+
+For complex element types, it computes a complex multiplication, with corner cases TBD.
+
+### Operands
+
+| Name  | Type                                                        |
+|-------|-------------------------------------------------------------|
+| `lhs` | tensor of integer, floating-point, or complex element types |
+| `rhs` | tensor of integer, floating-point, or complex element types |
```

Comment:
[shape_inference.cc](https://github.com/tensorflow/tensorflow/blob/475c1da0dcd0a3b13b5631b1ded8aaf16802bd01/tensorflow/compiler/xla/service/shape_inference.cc#L1011-L1016) supports multiplication for booleans as well. [Our ODS](https://github.com/openxla/stablehlo/blob/1115e9c8aac5d40168decae8b44166b2cb769ac3/stablehlo/dialect/StablehloOps.td#L766-L781) does as well. What should we do about this?